USER_GUIDE.md

```diff
@@ -169,6 +174,243 @@ for (int i = 0; i < searchResponse.hits().hits().size(); i++) {
+FieldSuggester fieldSuggester = new FieldSuggester.Builder().prefix("foo")
+                .completion(completionSuggester)
+                .build();
+Suggester suggester = new Suggester.Builder()
+                .suggesters(Collections.singletonMap(suggesterName, fieldSuggester))
+                .build();
+SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(index)
+                .suggest(suggester)
+                .build();
+
+SearchResponse<AppData> response = client.search(searchRequest, AppData.class);
+```
+
+### Using term suggester
+
+```java
+ String index = "term-suggester";
+        
+// term suggester does not require a special mapping
+client.indices().create(c -> c
+                .index(index));
+
+AppData appData = new AppData();
+appData.setIntValue(1337);
+appData.setMsg("foo");
+
+client.index(b -> b
+                .index(index)
+                .id("1")
+                .document(appData)
+                .refresh(Refresh.True));
+
+appData.setIntValue(1338);
+appData.setMsg("foobar");
+
+client.index(b -> b
+                .index(index)
+                .id("2")
+                .document(appData)
+                .refresh(Refresh.True));
+
+String suggesterName = "msgSuggester";
+
+TermSuggester termSuggester = FieldSuggesterBuilders.term()
+                .field("msg")
+                .size(1)
+                .build();
+FieldSuggester fieldSuggester = new FieldSuggester.Builder().text("fool")
+                .term(termSuggester)
+                .build();
+Suggester suggester = new Suggester.Builder()
+                .suggesters(Collections.singletonMap(suggesterName, fieldSuggester))
+                .build();
+SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(index)
+                .suggest(suggester)
+                .build();
+
+SearchResponse<AppData> response = client.search(searchRequest, AppData.class);
+```
+
+### Using phrase suggester
+
+```java
+String index = "phrase-suggester";
+
+String settingsJson = 
+"""
+  {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "trigram": {
+            "type": "custom",
+            "tokenizer": "standard",
+            "filter": [
+              "lowercase",
+              "shingle"
+            ]
+          }
+        },
+        "filter": {
+          "shingle": {
+            "type": "shingle",
+            "min_shingle_size": 2,
+            "max_shingle_size": 3
+          }
+        }
+      }
+    }
+  }
+""";
+
+String mappingJson =
+"""
+{
+  "properties": {
+    "msg": {
+      "type": "text",
+      "fields": {
+        "trigram": {
+          "type": "text",
+          "analyzer": "trigram"
+        }
+      }
+    }
+  }
+}
+""";
+
+
+ObjectMapper mapper = new JsonMapper();
+JsonpMapper jsonpMapper = new JacksonJsonpMapper(mapper);
+IndexSettings settings;
+try (JsonParser settingsParser = new JacksonJsonpParser(mapper.createParser(settingsJson))) {
```

Comment:
Do you think we could replace that with `IndexSettings.Builder` and `TypeMapping.Builder`? We should not encourage using raw `Xxx._DESERIALIZER`s in the official guides I think.